src/repositories/prisma/prisma-message-repository.ts

```diff
@@ -0,0 +1,14 @@
+import { prisma } from "../../prisma/client";
+import { IMessageRepository, MessageCreateData } from "../message-repository";
+
+export class PrismaMessageRepository implements IMessageRepository {
+  async create({ message, targetDate, targetEmail }: MessageCreateData): Promise<void> {
+    await prisma.message.create({
+      data: {
+        message,
+        targetDate,
+        targetEmail,
+      },
+    });
+  }
+}
```

Comment:
Sugiro que receba o cliente do prisma via construtor ao invés de importar, isso facilita na hora de testar pois podemos passar um mock do cliente prisma.

```suggestion
import { PrismaClient } from '@prisma/client';
import { IMessageRepository, MessageCreateData } from '../message-repository';

export class PrismaMessageRepository implements IMessageRepository {
  constructor(private prisma: PrismaClient) {}

  async create({
    message,
    targetDate,
    targetEmail,
  }: MessageCreateData): Promise<void> {
    await this.prisma.message.create({
      data: {
        message,
        targetDate,
        targetEmail,
      },
    });
  }
}
```

Reply:
Lembrando que se essa mudança for aceita, também tem que alterar a forma como essa classe é utilizar, pois ela passa a receber um parâmetro.

Reply:
Acho que essa alteração seria meio irrelevante já que não estamos a funcionalidade de salvar no banco de dados, estamos apenas testando se a função de criação está sendo chamada com o SPY do repository da mensagem.

Reply:
Concordo com esse ponto de vista.